File: sdk/src/main/java/com/cxense/cxensesdk/CredentialsProvider.java
package com.cxense.cxensesdk;

import androidx.annotation.NonNull;

/**
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2018-09-17).
 */
public interface CredentialsProvider {
    /**
     * Returns username
     *
     * @return username
     */
    @NonNull
    default String getUsername() {
        return "";
    }

    /**
     * Returns Api key
     *
     * @return api key
     */
    @NonNull
    default String getApiKey() {
        return "";
    }

    /**
     * Returns persistent query id for pushing DMP Performance events without authorization
     *
     * @return persistent query id
     */
    @NonNull
    default String getDmpPushPersistentId() {
        return "";
    }
}

File: sdk/src/main/java/com/cxense/cxensesdk/CredentialsProvider.kt
package com.cxense.cxensesdk

/**
 * Interface for providing API credentials.
 * Note that SDK doesn't cache results from these methods, which allows you to dynamically change credentials without replacing provider
 */
interface CredentialsProvider {
    /**
     * Returns username
     *
     * @return username
     */
    fun getUsername(): String

    /**
     * Returns Api key
     *
     * @return api key
     */
    fun getApiKey(): String

    /**
     * Returns persistent query id for pushing DMP Performance events without authorization
     *
     * @return persistent query id
     */
    fun getDmpPushPersistentId(): String
}
